addons/mail/data/mail_channel_data.xml

```diff
@@ -8,6 +8,9 @@
             <field name="description">General announcements for all employees.</field>
         </record>
 
+        <!-- make sure general channel is pinned for all its members -->
```

Comment:
Feel free to add to the comment this is because of the many2many/pivot not using default from intermediary model as it should.

Reply:
But I think a better alternative might be to move this `write` into the method responsible of it, which is `_subscribe_users`.

Best would be if the method actually created channel partner instead of using many2many so that all defaults are taken into account properly.

Also make sure the method works fast enough with many users.